### PR TITLE
will fix Generate Code Method #79

### PR DESCRIPTION
**Pull request by DivadNojnarg**
Cleanup dirty code chunk.

### CODE REVIEW

**Review by nbenn — COMMENTED** [review]
The change is an improvement for sure. What I don't understand here is why the `type` setting per field

https://github.com/blockr-org/blockr/blob/8f9a24a26bfd0cc3fa143fa893cf76c657c0d4ed/R/field.R#L14-L15

cannot be used for this. This feels like the canonical piece of infrastructure to handle this. Do we somehow need to make that more capable to cover additional use-cases?

File: R/utils.R [hunk 1]
```diff
@@ -184,7 +184,7 @@ dropNulls <- function(x) {
 #'
 #' @keywords internal
 convert_block <- function(from = new_select_block, to, data, ...) {
-  block <- initialize_block(from(data, ...), data)
+  block <- initialize_block(from(data, type = "name", ...), data)
```
Comment (DivadNojnarg) [inline R/utils.R:187]: @nbenn: I suppose this is what you meant about field type.

File: R/utils.R [hunk 2]
```diff
@@ -137,7 +137,7 @@ type_trans <- function(x) {
 
   switch(attr(x, "type"),
     literal = res,
-    name = as.name(res)
+    name = unlst(lapply(res, as.name))
```
Comment (DivadNojnarg) [inline R/utils.R:140]: type_tans was not able to deal with values of length > 1 (only returned the first element). I made this change here to ensure as.name applies to any element of res.

File: R/utils.R [hunk 3]
```diff
@@ -184,8 +184,10 @@ dropNulls <- function(x) {
 #'
 #' @keywords internal
 convert_block <- function(from = new_select_block, to, data, ...) {
-  block <- initialize_block(from(data, type = "name", ...), data)
+  block <- initialize_block(from(data, ...), data)
   class(block)[[1]] <- sprintf("%s_block", deparse(substitute(to)))
+  # Change type to name since arrange/group_by don't work with literals
+  attr(block$columns, "type") <- "name"
```
Comment (DivadNojnarg) [inline R/utils.R:190]: changed field type from literal to name so that `type_trans` can do its work with arrange and group_by blocks.

Comment (nbenn) [reply R/utils.R:190]: This has nothing to the with the PR, so feel free to ignore for now.

I really think we should get rid of `convert_block()`. If several blocks share functionality that we do not want to duplicate, we can always abstract this into functions and use those in multiple places.

File: R/utils.R (hunk 2 above)
Comment (nbenn) [reply R/utils.R:140]: Generally I feel that wrapping things with `unlist()` to make things work leads to other problems.

We might want to handle more explicitly the case where `value(x)` for a field is vector-valued.

Where does this cause issues? For list fields? I assume a range field would also not have a scalar `value()`, but you'd only want to use that as a literal in an expression.

I'm fine with just merging and sorting out this separately though.

**Review by nbenn — COMMENTED** [review]
I'm fine with merging. We can sort out the issues separately.

File: R/utils.R (hunk 3 above)
Comment (DivadNojnarg) [reply R/utils.R:190]: Agree. When we split block into core and extension, we can rework this.

**Comment by DivadNojnarg** [comment]
@nbenn: it seems I need re-approval. It is still blocked.

**Review by nbenn — COMMENTED** [review]
Lgtm.

**Review by nbenn — APPROVED** [review]
All good.

**Pull request merged**